py/server/deephaven/table_listener.py

```diff
@@ -72,208 +257,164 @@ def _do_locked(f: Callable, lock_type="shared") -> None:
-            elif nargs == 4:
-                listener_adapter = jpy.get_type(
-                    "io.deephaven.integrations.python"
-                    ".PythonReplayShiftObliviousListenerAdapter"
-                )
-                listener_adapter = listener_adapter(
-                    description, t.j_table, retain, listener
-                )
-            else:
-                raise ValueError(
-                    "Legacy listener must take 3 (added, removed, modified) or 4 (isReplay, added, removed, modified) "
-                    "arguments."
-                )
-
-        elif listener_type == "shift_aware":
-            if nargs == 1:
-                if replay_initial:
-                    raise ValueError(
-                        "ShiftObliviousListener does not support replay: ltype={} nargs={}".format(
-                            listener_type, nargs
-                        )
-                    )
-
-                listener_adapter = jpy.get_type(
-                    "io.deephaven.integrations.python.PythonListenerAdapter"
-                )
-                listener_adapter = listener_adapter(
-                    description, t.j_table, retain, listener
-                )
-            elif nargs == 2:
-                listener_adapter = jpy.get_type(
-                    "io.deephaven.integrations.python.PythonReplayListenerAdapter"
-                )
-                listener_adapter = listener_adapter(
-                    description, t.j_table, retain, listener
-                )
-            else:
-                raise ValueError(
-                    "Shift-aware listener must take 1 (update) or 2 (isReplay, update) arguments."
-                )
 
-        else:
-            raise ValueError(
-                "Unsupported listener type: ltype={}".format(listener_type)
-            )
+class TableListenerHandle:
+    """A handle for a table listener to manage its lifecycle."""
 
-        handle = TableListenerHandle(t, listener_adapter)
+    def __init__(self, t: Table, listener: Union[Callable, TableListener], description: str = None):
+        """Creates a new table listener handle.
 
-        def start():
-            if replay_initial:
-                listener_adapter.replay()
+        Table change events are processed by 'listener', which can be either
+        (1) a callable (e.g. function) or
+        (2) an instance of TableListener type which provides an "on_update" method.
 
-            if start_listening:
-                handle.register()
+        The callable or the on_update method must have one of the following two signatures.
+        * (update: TableUpdate, is_replay: bool): support replaying the initial table snapshot and normal table updates
+        The 'update' parameter is an object that describes the table update;
+        The 'is_replay' parameter is used only by replay listeners, it is set to 'true' when replaying the initial
+            snapshot and 'false' during normal updates.
```

Comment:
I think this is inaccurate given the current code.  E.g. only one signature is supported.